hydromt/models/root.py

```diff
@@ -0,0 +1,67 @@
+"""the new model root class."""
+
+from logging import Logger, getLogger
+from os import PathLike
+from os.path import isdir
+from pathlib import Path
+
+from hydromt.io.path import make_path_abs_and_cross_platform
+from hydromt.typing import ModeLike, ModelMode
+
+logger = getLogger(__name__)
+
+
+class ModelRoot:
+    """A class to handle model roots in a cross platform manner."""
+
+    def __init__(
+        self,
+        path: PathLike,
+        mode: ModeLike = "w",
+        logger: Logger = logger,
+    ):
+        self.mode = mode
+        self.path = path
+        self.logger: Logger = logger
+
+    @property
+    def path(self) -> Path:
+        """The path to the model root in question."""
+        return self._path
+
+    @path.setter
+    def path(self, path: PathLike) -> Path:
```

Comment:
In the current implementation setting the model path also updates the logger filehandler. We have workflows where you read a model, then change the root to a new output path, update and write the model. In that case the logger should, after reading, print to the new output path.